scripts/bottleneck_detector.py

```
@@ -0,0 +1,583 @@
+    alert_type: str  # "long_running_step", "queue_backlog", "resource_contention", "dependency_block"
+    severity: str  # "low", "medium", "high", "critical"
+    description: str
+    affected_steps: List[str]
+    affected_agents: List[str]
+    timestamp: float
+    resolution_suggestion: str = ""
+    metadata: Dict[str, Any] = field(default_factory=dict)
+
+
+class BottleneckDetector:
+    def __init__(self, detection_file: Path = None, max_history: int = 1000):
```

Ruff (0.14.4)

PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)